ProjFS.Mac/PrjFSKextTests/VnodeCacheTests.mm

```diff
@@ -0,0 +1,135 @@
+#import <XCTest/XCTest.h>
+
+typedef int16_t VirtualizationRootHandle;
+
+#include "../PrjFSKext/VnodeCacheTestable.hpp"
+#include "../PrjFSKext/VnodeCachePrivate.hpp"
+
+@interface VnodeCacheTests : XCTestCase
+@end
+
+@implementation VnodeCacheTests
+
+// Dummy vfs_context implementation for vfs_context_t
+struct vfs_context
+{
+};
+
+// Dummy vnode implementation for vnode_t
+struct vnode
+{
+    int dummyData;
+};
+
+// Dummy PerfTracer implementation for PerfTracer*
+class PerfTracer
+{
+};
+
+static vnode TestVnode;
+static const VirtualizationRootHandle TestRootHandle = 1;
+
+- (void)testComputeVnodeHashKeyWithCapacityOfOne {
+    s_entriesCapacity = 1;
+    vnode testVnode2;
+    vnode testVnode3;
+    
+    XCTAssertTrue(0 == ComputeVnodeHashKey(&TestVnode));
+    XCTAssertTrue(0 == ComputeVnodeHashKey(&testVnode2));
+    XCTAssertTrue(0 == ComputeVnodeHashKey(&testVnode3));
+}
+
+- (void)testInvalidateCache_ExclusiveLocked_SetsMemoryToZeros {
+    s_entriesCapacity = 100;
+    VnodeCacheEntry* emptyArray = static_cast<VnodeCacheEntry*>(calloc(s_entriesCapacity, sizeof(VnodeCacheEntry)));
+    s_entries = static_cast<VnodeCacheEntry*>(calloc(s_entriesCapacity, sizeof(VnodeCacheEntry)));
+    
+    memset(s_entries, 1, s_entriesCapacity * sizeof(VnodeCacheEntry));
+    XCTAssertTrue(0 != memcmp(emptyArray, s_entries, sizeof(VnodeCacheEntry) * s_entriesCapacity));
+    
+    InvalidateCache_ExclusiveLocked();
```

Comment:
InvalidateCache_ExclusiveLocked instead of VnodeCache_InvalidateCache, to explicitly skip locking.  If so thats fine a comment would be helpful.

Reply:
This test is actually to test `InvalidateCache_ExclusiveLocked` specifically.  I will work on a `VnodeCache_InvalidateCache` test as well.

Reply:
Does the test for the inner function really add anything we can't already check in the test for the outer one?